### PR TITLE
Panzer: add Quad8 to Quad4 mesh factory

### PR DESCRIPTION
**Pull request by rppawlo**
@trilinos/<teamName>

### CODE REVIEW

**Comment by tkordenbrock** [comment]
LGTM except for the shadow declaration mentioned above.

**Comment by rppawlo** [comment]
@etphipp - Thanks for reviewing the nox PR! Would you mind reviewing this one as well? Thanks!

**Review by tkordenbrock — APPROVED** [review]
LGTM

**Review by etphipp — APPROVED** [review]

File: packages/panzer/adapters-stk/src/stk_interface/Panzer_STK_Quad8ToQuad4MeshFactory.cpp [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+    std::vector<std::string> nodeset_names;
+    quad8Mesh_->getNodesetNames(nodeset_names);
+    for (const auto& n : nodeset_names)
+      mesh.addNodeset(n);
+  }
+
+  if(createEdgeBlocks_) {
+    const CellTopologyData * edge_ctd = shards::CellTopology(ctd).getBaseCellTopologyData(1,0);
+    std::vector<std::string> element_block_names;
+    quad8Mesh_->getElementBlockNames(element_block_names);
```
[the remark this answers is not in the document]
Comment (rppawlo) [reply packages/panzer/adapters-stk/src/stk_interface/Panzer_STK_Quad8ToQuad4MeshFactory.cpp:240]: Thanks @tkordenbrock - I'll take care of this in another PR shortly.

**Pull request merged**